app/controllers/support_interface/providers_controller.rb

```diff
@@ -21,7 +21,11 @@ def show
 
     def courses
       @provider = Provider.includes(courses: [:accredited_provider]).find(params[:provider_id])
-      @courses = @provider.courses.includes(accredited_provider: [:provider_agreements]).order(:name).group_by(&:recruitment_cycle_year)
+
+      courses = @provider.courses.includes(accredited_provider: [:provider_agreements]).order(:name).group_by(&:recruitment_cycle_year)
+      years = RecruitmentCycle.years_visible_in_support.each_with_object({}) { |year, hash| hash[year] = [] } # rubocop:disable Rails/IndexWith
```

Comment:
Would we consider moving this into a private method?

Reply:
I wouldn't think it's necessary until it's behaviour that needs to be shared. The fact that it's in line in that action tells it's own story, that it is specific to that action. That's my point of view at least.